packages/web/src/dev/ValidatorsListRow.tsx

```diff
@@ -87,23 +94,55 @@ class ValidatorsListRow extends React.PureComponent<Props & I18nProps, State> {
       document.removeEventListener('click', onDocumentClick, false)
   }
 
+  componentDidMount() {
+    this.setState({ isPinned: this.isPinned() })
+  }
+
   componentWillUnmount() {
     this.removeDocumentListener()
   }
 
+  isPinned(toggle?: boolean) {
+    const { address } = this.props.group
+    let list = (localStorage.getItem(localStoragePinnedKey) || '').split(',') || []
+    let isPinned = list.includes(address)
+    if (toggle) {
+      if (!isPinned) {
+        list.push(address)
+      } else {
+        list = list.filter((_) => _ !== address)
+      }
+      isPinned = !isPinned
+      localStorage.setItem(localStoragePinnedKey, list.join(','))
+      this.props.onPinned()
+    }
+    return isPinned
+  }
+
   render() {
     const { group, expanded } = this.props
-    const { tooltip } = this.state
-    const stopPropagation = (event: any) => {
+    const { tooltip, isPinned } = this.state
+    const stopPropagation = (event) => {
       event.stopPropagation()
     }
-    const toggleTooltip = (event: any) => {
+    const toggleTooltip = (event) => {
       stopPropagation(event)
       this.setState({ tooltip: !tooltip })
     }
+    const togglePinned: any = (event) => {
```

Comment:
im having trouble understanding why this and the  `toggleTooltip` above are defined in render? 

same with the `stopPropagation`.

Reply:
I can move it to a method, but I'll need to create a new function inside render or bind the context to the method to be able to use `this` context

Reply:
if you want a method to be alwasy lexicaly bound to its class you just need to define it an arrow function on the class like this

```
class ValidatorsListRow extends React.PureComponent<Props & I18nProps, State> {

togglePinned = () => {
```